app.js

```diff
@@ -0,0 +1,928 @@
+			chai.assert.equal(list[0].price, 1)
+			chai.assert.equal(list[1].item, 'test2')
+			chai.assert.equal(list[1].price, 2)
+		});
+
+		it('should remove i number from START if list if i > 0', () => {
+			let list = addToShoppingList({
+				'item': 'test',
+				'price': 1
+			});
+			list = addToShoppingList({
+				'item': 'test2',
+				'price': 2
+			}, list);
+			list = addToShoppingList({
+				'item': 'test3',
+				'price': 3
+			}, list);
+
+			list = smartRemoveItems(1, list);
+
+			chai.assert.equal(list.length, 2)
+			chai.assert.equal(list[0].item, 'test2')
+			chai.assert.equal(list[0].price, 2)
+			chai.assert.equal(list[1].item, 'test3')
+			chai.assert.equal(list[1].price, 3)
+		});
+	});
+
+	/* 8
+		@function spliceItem
+		@para item {object}
+		@param i {number}
+		@param list {array, []}
+		@returns list
+		@description
+			- item must be an object that looks like this:
+			{
+				'item': 'eggs',
+				'price': 1.59
+			} (else throw error)
+			- insert item into the ith index of the list
+			- if i > length of list, just append
+			- if i < 0, just prepend
+	*/
+
+	// implement function here
+	const spliceItem = (item, i, list = []) => {
+		if(!item.hasOwnProperty('item') || !item.hasOwnProperty('price')){
+			throw new Error('This is not valid')
+		} else if(i > list.length) {
+			list.push(item);
+		} else if(i < 0) {
+			list.unshift(item);
+		}
+	}
+
+	// TEST
+	describe('8. spliceItem', () => {
+		it('should throw an error if item is not valid', () => {
+			chai.assert.throws(() => {
+				spliceItem('invalidItem', 0, [])
+			}, Error);
+		});
+
+		it('should insert item to the ith index of the list', () => {
+			const list = spliceItem({
+				'item': 'test',
+				'price': 1,
+			}, 0, [])
+
+			chai.assert.equal(list[0].item, 'test')
+			chai.assert.equal(list[0].price, 1)
+		});
+
+		it('should append to the end if i > length of list', () => {
+			const list = spliceItem({
+				'item': 'test',
+				'price': 1,
+			}, 9, [{
+				'item': 'test0',
+				'price': 0,
+			}])
+
+			chai.assert.equal(list[1].item, 'test')
+			chai.assert.equal(list[1].price, 1)
+		});
+
+		it('should prepend if i < 0', () => {
+			const list = spliceItem({
+				'item': 'test',
+				'price': 1,
+			}, -1, [{
+				'item': 'test0',
+				'price': 0,
+			}])
+
+			chai.assert.equal(list[0].item, 'test')
+			chai.assert.equal(list[0].price, 1)
+		});
+
+	});
+
+	/* 9
+		@function spliceItems
+		@param items {list}
+		@param i {number}
+		@param list {array, []}
+		@returns list
+		@description
+			- *EACH* item in `items` must be an object 
+			that looks like this:
+			{
+				'item': 'eggs',
+				'price': 1.59
+			} (else throw error)
+			- insert items into the ith index of the list
+			- if i > length of list, just append
+			- if i < 0, just prepend
+			- if `items` is empty, return list
+	*/
+
+	// implement function here
+	const spliceItems = (items, i, list = []) => {
+		if(typeof item !== 'string' || typeof price !== 'number') {
```

Comment:
Good try, but we need to evaluate _each_ item in the list, whereas this logic is only valid for one single item `item`.